src/pages/library/datasetBuilder/CohortEditor.ts

```diff
@@ -0,0 +1,428 @@
+  // This needs to be replaced with a UI that lets users select the criteria they want from
+  // the list of concepts for this domain.
+  return {
+    domainType,
+    name: domainType.values[0],
+    id: criteriaCount++,
+    // Need to call the API service to get the count for this criteria.
+    count: 100,
+  };
+};
+
+const createDefaultListCriteria = (listType: ProgramDataListType): ProgramDataListCriteria => {
+  return {
+    listType,
+    name: listType.name,
+    id: criteriaCount++,
+    count: 100,
+    value: listType.values[0],
+  };
+};
+
+const createDefaultRangeCriteria = (rangeType: ProgramDataRangeType): ProgramDataRangeCriteria => {
+  return {
+    rangeType,
+    name: rangeType.name,
+    id: criteriaCount++,
+    count: 100,
+    low: rangeType.min,
+    high: rangeType.max,
+  };
+};
+
+export function createCriteriaFromType(type: CriteriaType): AnyCriteria {
+  return (
+    Utils.condTyped<AnyCriteria>(
+      ['category' in type, () => selectDomainCriteria(type as DomainType)],
+      ['values' in type, () => createDefaultListCriteria(type as ProgramDataListType)],
+      ['min' in type, () => createDefaultRangeCriteria(type as ProgramDataRangeType)]
+    ) ?? { domainType: { id: 0, category: 'unknown', values: [] }, name: 'unknown', count: 0, id: 0 }
+  );
+}
+
+type CriteriaGroupViewProps = {
+  index: number;
+  criteriaGroup: CriteriaGroup;
+  updateCohort: CohortUpdater;
+  cohort: Cohort;
+  datasetDetails: DatasetResponse;
+};
+
+const CriteriaGroupView = ({ index, criteriaGroup, updateCohort, cohort, datasetDetails }: CriteriaGroupViewProps) => {
+  return div(
+    {
+      // key: criteriaGroup.name,
+      style: {
+        backgroundColor: 'white',
+        borderRadius: '5px',
+        marginTop: 5,
+        border: `1px solid ${colors.dark(0.35)}`,
+      },
+    },
+    [
+      div({ style: { padding: '1rem' } }, [
+        div(
+          {
+            style: {
+              display: 'flex',
+              width: '100%',
+              justifyContent: 'space-between',
+              alignItems: 'baseline',
+            },
+          },
+          [
+            div(
+              {
+                style: {
+                  display: 'flex',
+                  justifyContent: 'space-between',
+                  alignItems: 'baseline',
+                },
+              },
+              [
+                h(Select, {
+                  'aria-label': 'must or must not meet',
+                  options: ['Must', 'Must not'],
+                  value: criteriaGroup.mustMeet ? 'Must' : 'Must not',
+                  onChange: () => updateCohort(_.set(`criteriaGroups.${index}.mustMeet`, !criteriaGroup.mustMeet)),
+                }),
+                div({ style: { margin: '0 10px' } }, ['meet']),
+                h(Select, {
+                  'aria-label': 'all or any',
+                  styles: { container: (provided) => ({ ...provided, style: { marginLeft: 10 } }) },
```

Comment:
Opt nit: If you wanted you could pull the `10` out to a const and reuse it on the other divs in this row.